### PR TITLE
Replace realpath with Path.expand

### PR DESCRIPTION
**Pull request by mickel8**
`realpath` might not be installed on some OSes. I wonder if we could just use `Path.expand` instead, like in the PR?

### CODE REVIEW

File: lib/bundlex/toolchain/common/unix/os_deps.ex [hunk 1]
```diff
@@ -140,8 +140,7 @@ defmodule Bundlex.Toolchain.Common.Unix.OSDeps do
     # are stored because we need to assure that the relative link in the symlink's target
     # will work properly also when the symlink is put in the location, for which the
     # logical path differs from the physical path
-    {physical_output_path, 0} = System.shell("realpath #{logical_output_path}")
-    physical_output_path = String.trim_trailing(physical_output_path)
+    physical_output_path = logical_output_path |> Path.expand() |> String.trim_trailing()
```
Comment (FelonEkonom) [inline lib/bundlex/toolchain/common/unix/os_deps.ex:143]:
```suggestion
{physical_output_path, 0} = 
  try do
   System.cmd("realpath", ["logical_output_path"])
  rescue
    e in ErlangError -> 
      with %{original: :enoent} <- e do
        Logger.error("realpath is required by Bundlex, but it is not installed")
      end
      
      reraise e, __STACKTRACE__
    end    
    
  physical_output_path = String.trim_trailing(physical_output_path)
```
Is it ok for you?

Comment (mickel8) [reply lib/bundlex/toolchain/common/unix/os_deps.ex:143]: It's for sure better! :)

**Review by mat-hek — CHANGES_REQUESTED** [review]
Since it seems that the lack of `realpath` is the only thing that currently prevents Bundlex from working on a clean Mac OS, I'd instead use `:file.read_link`. Since it works only for symlinks, it would probably require having `Bundlex.Toolchain.real_output_path` that would check if `priv` is a symlink and resolve it's real path in that case.

**Comment by mat-hek** [comment]
@mickel8 it should work without `realpath` now, can you check?

File: test/bundlex/integration_test.exs [hunk 2]
```diff
@@ -1,71 +1,78 @@
...
-    target_path = "#{project}_moved"
-    File.rename!(project, target_path)
-
-    on_exit(fn ->
-      File.rename("#{project}_moved", project)
-    end)
-
-    target_path
-  end
+  Enum.map([false, true], fn priv ->
```
Comment (FelonEkonom) [inline test/bundlex/integration_test.exs:2]:
```suggestion
  Enum.map([false, true], fn priv? ->
```

**Review by FelonEkonom — APPROVED** [review]

**Comment by mickel8** [comment]
It works, the only things is taht `/bin/sh: realpath: command not found` is printed

**Review by mat-hek — APPROVED** [review]

**Pull request merged**